src/server/document-controller.spec.js

```diff
@@ -0,0 +1,92 @@
+import sinon from 'sinon';
+import httpErrors from 'http-errors';
+import { EventEmitter } from 'events';
+import httpMocks from 'node-mocks-http';
+import uniqueId from '../utils/unique-id.js';
+import DocumentController from './document-controller.js';
+import { setupTestUser, setupTestEnvironment, destroyTestEnvironment, pruneTestEnvironment } from '../test-helper.js';
+
+const { BadRequest } = httpErrors;
+
+describe('document-controller', () => {
+  const sandbox = sinon.createSandbox();
+  let container;
+
+  let documentService;
+  let clientDataMapper;
+  let pageRenderer;
+
+  let user;
+  let req;
+  let res;
+  let sut;
+
+  beforeAll(async () => {
+    container = await setupTestEnvironment();
+    user = await setupTestUser(container);
+  });
+
+  afterAll(async () => {
+    await destroyTestEnvironment(container);
+  });
+
+  beforeEach(() => {
+    documentService = {
+      createRegenerateDocumentsBatch: sandbox.stub()
+    };
+
+    sut = new DocumentController(documentService, clientDataMapper, pageRenderer);
+  });
+
+  afterEach(async () => {
+    await pruneTestEnvironment(container);
+    sandbox.restore();
```

Comment:
You should reverse those lines, because `pruneTestEnvironment` might otherwise use still mocked functions

Reply:
Another thing: we usually don't write tests that go to the db on the controller level (except for `client-data-mapper`), so the test environment isn't really needed here, we usually just unit test the controllers and mock the called services.

Reply:
I removed the test environment, thanks.